consensus/src/counters.rs

```diff
@@ -248,6 +248,89 @@ pub static CHAIN_HEALTH_PARTICIPATING_NUM_VALIDATORS: Lazy<Vec<IntGauge>> = Lazy
         .collect()
 });
 
+/// Emits consensus participation status for all peers, 0 means no participation in the window
+/// 1 otherwise.
+pub static CONSENSUS_PARTICIPATION_STATUS: Lazy<IntGaugeVec> = Lazy::new(|| {
+    register_int_gauge_vec!(
+        "aptos_consensus_participation_status",
+        "Counter for consensus participation status, 0 means no participation and 1 otherwise",
+        &["peer_id"]
+    )
+    .unwrap()
+});
+
+/// Voting power of the validator
+pub static VALIDATOR_VOTING_POWER: Lazy<Gauge> = Lazy::new(|| {
+    register_gauge!(
+        "aptos_validator_voting_power",
+        "Voting power of the validator"
+    )
+    .unwrap()
+});
+
+/// Emits voting power for all validators in the current epoch.
+pub static ALL_VALIDATORS_VOTING_POWER: Lazy<IntGaugeVec> = Lazy::new(|| {
+    register_int_gauge_vec!(
+        "aptos_all_validators_voting_power",
+        "Voting power for all validators in current epoch",
+        &["peer_id"]
+    )
+    .unwrap()
+});
+
+/// For the current ordering round, voting power needed for quorum.
+pub static CONSENSUS_CURRENT_ROUND_QUORUM_VOTING_POWER: Lazy<Gauge> = Lazy::new(|| {
+    register_gauge!(
+        "aptos_consensus_current_round_quorum_voting_power",
+        "Counter for consensus participation status, 0 means no participation and 1 otherwise",
+    )
+    .unwrap()
+});
+
+/// For the current ordering round, for each peer, whether they have voted, and for which hash_index
+pub static CONSENSUS_CURRENT_ROUND_VOTED_POWER: Lazy<GaugeVec> = Lazy::new(|| {
+    register_gauge_vec!(
+        "aptos_consensus_current_round_voted_power",
+        "Counter for consensus participation status, 0 means no participation and 1 otherwise",
+        &["peer_id", "hash_index"]
+    )
+    .unwrap()
+});
+
+/// For the current ordering round, for each peer, whether they have voted for a timeout
+pub static CONSENSUS_CURRENT_ROUND_TIMEOUT_VOTED_POWER: Lazy<GaugeVec> = Lazy::new(|| {
+    register_gauge_vec!(
+        "aptos_consensus_current_round_timeout_voted_power",
+        "Counter for consensus participation status, 0 means no participation and 1 otherwise",
+        &["peer_id"]
+    )
+    .unwrap()
+});
+
+/// Since we cannot have metric be a tuple, have round be represented as epoch * multiplier + round
+/// Use number that is larger than commonly used epochs.
+pub static EPOCH_ROUND_MULTIPLIER: u64 = 1_000_000;
```

Comment:
May be bump it up to 1B instead of 1M, so that it leaves enough room for longer epoch length of several days. 1M is cutting too close for an epoch length of 1 day.

Reply:
are we ever going to have an epoch of 1 day? :)